be/src/Project/Web/Buyer/Buyer.Web/Shared/Repositories/MainNavigationLinks/MainNavigationLinkRepository.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Buyer.Web.Shared.DomainModels.MainNavigationLinks;
+using Buyer.Web.Shared.GraphQlResponseModels;
+using Foundation.Extensions.ExtensionMethods;
+using GraphQL;
+using GraphQL.Client.Abstractions;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Buyer.Web.Shared.Repositories.MainNavigationLinks
+{
+    public class MainNavigationLinkRepository : IMainNavigationLinkRepository
+    {
+        private readonly IGraphQLClient _graphQlClient;
+        private readonly ILogger<MainNavigationLinkRepository> _logger;
+
+        public MainNavigationLinkRepository(
+            IGraphQLClient graphQlClient,
+            ILogger<MainNavigationLinkRepository> logger)
+        {
+            _graphQlClient = graphQlClient;
+            _logger = logger;
+        }
+
+        public async Task<IEnumerable<MainNavigationLink>> GetMainNavigationLinksAsync(string contentPageKey, string language, string fallbackLanguage)
+        {
+            try
+            { 
+                var response = await _graphQlClient.SendQueryAsync<JObject>(GetMainNavigationLinksQuery(contentPageKey, language));
+
+                if (response.Errors.OrEmptyIfNull().Any() is false && response?.Data != null)
```

Comment:
**suggestion:** I think it might be simplified like it is on HeroSlider

![obraz](https://github.com/dawiddworak88/Giuru/assets/93117743/2b844216-a6db-48b9-85d3-be7840f37657)

Reply:
In HeroSliderRepository is possible to simple code because it is only for HomePage. MainNavigationLink is used in many pages so i need replaced contentPageKey to "page" to prevent redundant GraphQlResponsModel.